ccd/deployment.yaml

```diff
@@ -0,0 +1,825 @@
+#
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: service-auth-provider-api
+  namespace: ccd
+data:
+  SERVER_PORT: '8080'
+  JAVA_OPTS: '-Xms8m -Xmx256m -Djava.security.egd=file:/dev/./urandom -XX:MaxMetaspaceSize=128m'
+  JWT_KEY: 'wThK0f0/lh3FlxFcL4xUWDMI5C1J9KyQBgXV4wseh1e5J1uYJIjvTvArHxQDrYoHJ23xFxjHkOnvNbR5dXRoxA=='
+  MICROSERVICE_KEYS_BULK_SCAN_ORCHESTRATOR: 'AAAAAAAAAAAAAAAA'
+  MICROSERVICE_KEYS_EM_GW: 'AAAAAAAAAAAAAAAA'
+  MICROSERVICE_KEYS_CCD_DATA: 'AAAAAAAAAAAAAAAA'
+  MICROSERVICE_KEYS_CCD_DEFINITION: 'AAAAAAAAAAAAAAAA'
+  MICROSERVICE_KEYS_CCD_GW: 'AAAAAAAAAAAAAAAA'
+  TESTING_SUPPORT_ENABLED: 'true'
+---
+#
+# CCD-USER-PROFILE-API CONFIG
+#
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ccd-user-profile-api
+  namespace: ccd
+data:
+  JAVA_OPTS: '-Xms8m -Xmx256m -Djava.security.egd=file:/dev/./urandom -XX:MaxMetaspaceSize=128m'
+  USER_PROFILE_DB_HOST: 'shared-database'
+  USER_PROFILE_DB_PORT: '5432'
+  USER_PROFILE_DB_USERNAME: 'ccd_user_profile'
+  USER_PROFILE_DB_PASSWORD: 'ccd_user_profile'
+  USER_PROFILE_S2S_AUTHORISED_SERVICES: 'ccd_data,ccd_definition'
+  IDAM_S2S_URL: 'http://service-auth-provider-api'
+  REFORM_SERVICE_NAME: 'ccd-user-profile-api'
+  REFORM_TEAM: 'ccd'
+  REFORM_ENVIRONMENT: 'local'
+  APPINSIGHTS_INSTRUMENTATIONKEY: 'fake-key'
+  USER_PROFILE_DB_USE_SSL: 'false'
+---
+#
+# CCD-DEFINITION-STORE-API CONFIG
+#
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ccd-definition-store-api
+  namespace: ccd
+data:
+  JAVA_OPTS: '-Xms8m -Xmx256m -Djava.security.egd=file:/dev/./urandom -XX:MaxMetaspaceSize=128m'
+  DEFINITION_STORE_DB_HOST: 'shared-database'
+  DEFINITION_STORE_DB_PORT: '5432'
+  DEFINITION_STORE_DB_USERNAME: 'ccd_definition_store'
+  DEFINITION_STORE_DB_PASSWORD: 'ccd_definition_store'
+  SPRING_DATASOURCE_URL: 'jdbc:postgresql://shared-database:5432/ccd_definition_store'
+  DEFINITION_STORE_IDAM_KEY: 'AAAAAAAAAAAAAAAA'
+  DEFINITION_STORE_S2S_AUTHORISED_SERVICES: 'ccd_data,bulk_scan_orchestrator,ccd_gw'
+  USER_PROFILE_HOST: 'http://ccd-user-profile-api'
+  IDAM_USER_URL: 'http://idam-api'
+  IDAM_S2S_URL: 'http://service-auth-provider-api'
+  REFORM_SERVICE_NAME: 'ccd-definition-store-api'
+  REFORM_TEAM: 'ccd'
+  REFORM_ENVIRONMENT: 'local'
+  APPINSIGHTS_INSTRUMENTATIONKEY: 'fake-key'
+---
+#
+# CCD-DATA-STORE-API CONFIG
+#
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ccd-data-store-api
+  namespace: ccd
+data:
+  JAVA_OPTS: '-Xms8m -Xmx256m -Djava.security.egd=file:/dev/./urandom -XX:MaxMetaspaceSize=128m'
+  DATA_STORE_DB_HOST: 'shared-database'
+  DATA_STORE_DB_PORT: '5432'
+  DATA_STORE_DB_USERNAME: 'ccd_data_store'
+  DATA_STORE_DB_PASSWORD: 'ccd_data_store'
+  DATA_STORE_IDAM_KEY: 'AAAAAAAAAAAAAAAA'
+  SPRING_DATASOURCE_URL: 'jdbc:postgresql://shared-database:5432/ccd_data_store?stringtype=unspecified'
+  DATA_STORE_TOKEN_SECRET: 'sdsadahjsadhgaskjhdhasghgkjasd'
+  DATA_STORE_S2S_AUTHORISED_SERVICES: 'bulk_scan_orchestrator,ccd_gw'
```

Comment:
extension point for teams to add their own services, ccd_gw always need to be there but teams will have other ones